dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -378,8 +378,10 @@ public Workflow setupWorkflowVersions(String repositoryId, Workflow workflow, Op
             GHRef[] refs = repository.getRefs();
             for (GHRef ref : refs) {
                 Triple<String, Date, String> referenceTriple = getRef(ref, repository);
-                if (versionName.isEmpty() || Objects.equals(versionName.get(), referenceTriple.getLeft())) {
-                    references.add(referenceTriple);
+                if (referenceTriple != null) {
```

Comment:
Don't change this, since it's now working, but this could have been a case to use Optional, to force the caller to check.